ci-operator/step-registry/azure/provision/custom-role/azure-provision-custom-role-commands.sh

```diff
@@ -297,16 +330,17 @@ ${required_permissions}
 """
     fi
 
-    # optional permissions when specifying UserAssigned identity
-    if [[ "${install_config_identity_user_default}" == "UserAssigned" ]] || [[ "${install_config_identity_user_master}" == "UserAssigned" ]] || [[ "${install_config_identity_user_compute}" == "UserAssigned" ]]; then
-        required_permissions="""
+
+    # Optional permissions when configuring identity type to UserAssigned with precreated user-assigend identity
+    if [[ "${master_identity_type}" == "UserAssigned" ]] && [[ -n "${master_user_identity}" ]] && [[ "${worker_identity_type}" == "UserAssigned" ]] && [[ -n "${worker_user_identity}" ]]; then
```

Comment:
if master identity type is none, worker identity type is UserAssigned + customer id, then the condition looks not right.
```suggestion
    if [[ "${master_identity_type}" == "UserAssigned" && -n "${master_user_identity}" ]] || [[ "${worker_identity_type}" == "UserAssigned" && -n "${worker_user_identity}" ]]; then
```

Reply:
hmm, right, but suggested check seems also not accurate, and might have duplicated roles in role definition with below scenario:
> master: UserAssigned type
worker: UserAssigned type + custom identity

maybe we can move this check to line320 as `elif` condition.

Reply:
okay for me, or we can run a `uniq` to remove the duplicates permissions for the whole permission list before using it.